appcues/src/main/java/com/appcues/debugger/screencapture/AndroidTargetingStrategy.kt

```diff
@@ -7,38 +7,52 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.compose.ui.semantics.SemanticsNode
 import androidx.compose.ui.semantics.SemanticsOwner
+import androidx.compose.ui.semantics.SemanticsProperties
 import androidx.compose.ui.semantics.SemanticsPropertyKey
 import androidx.compose.ui.semantics.SemanticsPropertyReceiver
 import androidx.core.view.children
 import com.appcues.ElementSelector
 import com.appcues.ElementTargetingStrategy
 import com.appcues.ViewElement
+import com.appcues.debugger.screencapture.AndroidViewSelector.Companion.SELECTOR_APPCUES_ID
+import com.appcues.debugger.screencapture.AndroidViewSelector.Companion.SELECTOR_CONTENT_DESCRIPTION
+import com.appcues.debugger.screencapture.AndroidViewSelector.Companion.SELECTOR_RESOURCE_NAME
+import com.appcues.debugger.screencapture.AndroidViewSelector.Companion.SELECTOR_TAG
 import com.appcues.isAppcuesView
 import com.appcues.monitor.AppcuesActivityMonitor
 
-internal data class AndroidViewSelector(
-    var contentDescription: String? = null,
-    var tag: String? = null,
-    var resourceName: String? = null
+internal class AndroidViewSelector(
+    private val properties: Map<String, String?>,
+    val type: String? = null,
 ) : ElementSelector {
 
+    companion object {
+
+        const val SELECTOR_CONTENT_DESCRIPTION = "contentDescription"
+        const val SELECTOR_TAG = "tag"
+        const val SELECTOR_RESOURCE_NAME = "resourceName"
+        const val SELECTOR_APPCUES_ID = "appcuesId"
+    }
+
+    private val contentDescription: String? = properties[SELECTOR_CONTENT_DESCRIPTION]
+    private val tag: String? = properties[SELECTOR_TAG]
+    private val resourceName: String? = properties[SELECTOR_RESOURCE_NAME]
+    private val appcuesId: String? = properties[SELECTOR_APPCUES_ID]
+
     val isValid: Boolean
         get() = contentDescription != null || tag != null || resourceName != null
 
     val displayName: String?
         get() = when {
-            resourceName != null -> "RES/$resourceName"
-            tag != null -> "TAG/$tag"
+            resourceName != null -> "$type ($resourceName)"
+            tag != null -> "$type (tag $tag)"
+            appcuesId != null -> appcuesId
             contentDescription != null -> contentDescription
             else -> null
```

Comment:
I think of `resourceName` analogous to `accessibilityIdentifier` on the [iOS side](https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Presentation/Debugger/ScreenCapturing/UIKitElementTargeting.swift#L85-L96)

So matching that pattern, how about
```kotlin
when {
    resourceName != null -> resourceName
    appcuesId != null -> appcuesId
    tag != null -> "$type (tag $tag)"
    contentDescription -> "$type ($contentDescription)"
    else -> null

}
```

Reply:
I will change that, I inverted resourceName with contentDescription there